### DISCUSSION TITLE
Can I use BlockSuite as plain code editor, or build a plain code editor based on code block components?

### DISCUSSION THREAD

**Discussion by alissa-tung** [toeverything/blocksuite#6552] [Q&A]
I am curious about did we already has code editor, which is like monaco, without some monaco fancy functionalities, just edit functionalities (redo, ...)? I noticed that we did already has code blocks. But I am not sure about whether I can use it as code editor.

There are components like page (doc) editor, and edgeless editor, but they are not simple as a code editor for plain text, and is not in style of monaco like or other code editor

**Comment by doouding** [comment]
I'm really surprised because what you said is exactly where we're headed. We want to have different types of editor to edit the same document. For example,  PDF editor could add annotations to a PDF block, and code editor could edit a code block.
Your idea totally works within our current tech architecture; but it requires writing quite a bit of code to integrate the Monaco editor with our data layer. We will provide some examples like that in the coming future.

**Reply by alissa-tung** [reply]
Thank you. I am curious about, does BlockSuite is going to write data layers around Monaco editor? While I am playing with BlockSuite, I had some tries on

rendering a document with code block, add a toggle on code block, on click is set to expand to a Monaco editor.

I am looking forward to seeing what approach will BlockSuite soon. Thank you for developing the library.
